### PR TITLE
Mac cmake support

### PR DESCRIPTION
**Pull request by FunMiles**
I added support to build the library and the samples using CMake.
Among other things, this allows to checkout the code with CLion and compile it directly out of the box.
I have tried to make sure that my modifications result in no changes if not used under Mac OS X but have no way to test all cases.

### CODE REVIEW

**Comment by louis-mclaughlin** [comment]
>I have tried to make sure that my modifications result in no changes if not used under Mac OS X

Tested this in Linux (Mint 17), works fine for me.

**Comment by FunMiles** [comment]
Thanks Louis. In the gameplay/CMakeLists.txt, there are the following two lines:

add_definitions(-std=c++11)
add_definitions(-lstdc++)

The second line is absurd, since it is a linking argument and not a source code compilation flag. It also causes some warning with clang++ on the Mac. It should be replaced, I believe, with:

add_definitions(-std=c++11 -stdlib=libc++)

But I am not sure if on linux or other platform clang++ is being used. If not, this should be protected by an 'if' testing whether the compiler is clang++ or not. And for g++, there shouldn't be any need to even specify -lstdc++ since it should always be linked in anyway. If it is necessary, however, it is not as a definition but as an added linked library.
Could you check if on linux, just removing the second line still compiles everything?

**Comment by louis-mclaughlin** [comment]
Sorry, I must have made a typo when I cloned and checked-out your branch as I have since discovered some errors, I've fixed them in this [pull request](https://github.com/FunMiles/GamePlay/pull/1) to your branch.

**Comment by louis-mclaughlin** [comment]
>Could you check if on linux, just removing the second line still compiles everything?

Done, everything builds and runs successfully using gcc but Clang fails to compile, I've fixed it in this [pull request](https://github.com/FunMiles/GamePlay/pull/2)

Also, since this CMake change is a new feature, you'll need to make a pull request against the 'next' branch rather than 'master'.

**Comment by seanpaultaylor** [comment]
Have you gone back and ensured with still works properly on the other existing platforms that were working for CMake ?

File: samples/CMakeLists.txt [hunk 1]
```diff
@@ -1,33 +1,76 @@
 include(BuildHelpers.CMakeLists.txt)
 
-include_directories( 
-    ${CMAKE_SOURCE_DIR}/gameplay/src
-    ${CMAKE_SOURCE_DIR}/external-deps/include
+include_directories(
+        ${CMAKE_SOURCE_DIR}/gameplay/src
```
Comment (seanpaultaylor) [inline samples/CMakeLists.txt]: Tab issue.

**Comment by seanpaultaylor** [comment]
If this does go in, it will be in next first and not master.

File: samples/CMakeLists.txt (hunk 1 above)
Comment (FunMiles) [inline samples/CMakeLists.txt]: I use CLion for editing and use their 'smart tab' formatting system. Sorry if my liberal use of auto formatting sometimes affects existing formatting that should not be changed.

**Comment by FunMiles** [comment]
Thanks Louis, I have taken your pull request and will move things over to the next branch.
I also have some changes for which I want to make a pull request to add the 'override' label on overriding virtual methods. Should that go to next or master?

**Pull request closed without merging**